src/System.Text.Json/src/System/Text/Json/Utf8JsonReader.MultiSegment.cs

```diff
@@ -2260,6 +2268,33 @@ private bool ConsumeSingleLineCommentMultiSegment(ReadOnlySpan<byte> localBuffer
                 }
             } while (idx == -1);
 
+            // If we have encountered \r, check and advance index if next character is \n
+            if (localBuffer[idx] == JsonConstants.CarriageReturn)
+            {
+                if (idx < localBuffer.Length - 1)
+                {
+                    if (localBuffer[idx + 1] == JsonConstants.LineFeed)
+                        idx++;
+                }
+                else if (!IsLastSpan)
+                {
+                    prevTotalConsumed = _totalConsumed;
+                    if (!GetNextSpan())
+                    {
+                        _totalConsumed = prevTotalConsumed;
```

Comment:
I don't think we should be returning false here.

What happens if the data ends with a single line comment?

Consuming it means the value needs to be returned to the caller.

Say we have: "{}// some comment\r"

```C#
// Let's say we allow comments.
jsonreader.Read() // start object, should return true
jsonreader.Read() // end object, should return true
jsonreader.Read() // comment value (// some comment)
jsonreader.Read() // now this should return false
```